test_common/harness/imageHelpers.cpp

```diff
@@ -874,9 +874,9 @@ void get_max_sizes(
             assert(x0 > 0 && M > 0);
 
             // Store the size
-            sizes[(*numberOfSizes)][fixed_dim] = M;
-            sizes[(*numberOfSizes)][x0_dim] = x0;
-            sizes[(*numberOfSizes)][x1_dim] = x1;
+            sizes[(*numberOfSizes)][fixed_dim] = static_cast<size_t>(M);
+            sizes[(*numberOfSizes)][x0_dim] = static_cast<size_t>(x0);
```

Comment:
Why do we need to cast here? Everything is already `size_t`. 
Slightly off-topic, the `x0` and `x1` variable names could be improved since they are not descriptive and look like hex literals.